barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -1695,7 +1748,7 @@ void AvmTraceBuilder::op_returndata_copy(uint8_t indirect,
 // Helper for "gas left" related opcodes
 void AvmTraceBuilder::execute_gasleft(EnvironmentVariable var, uint8_t indirect, uint32_t dst_offset)
 {
-    assert(var == EnvironmentVariable::L2GASLEFT || var == EnvironmentVariable::DAGASLEFT);
+    ASSERT(var == EnvironmentVariable::L2GASLEFT || var == EnvironmentVariable::DAGASLEFT);
```

Comment:
Related to assert vs. ASSERT: It does not change anything AFAIK, but the latter was usually the adopted style throughout our code base.

It is really an assertion and not error handling because this function is called only from the externally exposed routines: op_l2gasleft() and op_dagasleft

In other words, this is a coding mistake.

Related to `op_get_env_var`, it is guarded by the first if. the "default" case in the switch is to make the compiler happy but we have the guarantee that it cannot happen. I will add a comment about this.